core/src/stdlib/pyscript/events.py

```diff
@@ -92,60 +92,26 @@ def when(target, *args, **kwargs):
             elements = selector if isinstance(selector, list) else [selector]
 
     def decorator(func):
-        if config["type"] == "mpy":  # Is MicroPython?
+        sig = inspect.signature(func)
+        if sig.parameters:
             if is_awaitable(func):
 
-                async def wrapper(*args, **kwargs):
-                    """
-                    This is  a very ugly hack to get micropython working because
-                    `inspect.signature` doesn't exist. It may be actually better
-                    to not try any magic for now and raise the error.
-                    """
-                    try:
-                        return await func(*args, **kwargs)
-
-                    except TypeError as e:
-                        if "takes" in str(e) and "positional arguments" in str(e):
-                            return await func()
-                        raise
+                async def wrapper(event):
+                    return await func(event)
 
             else:
-
-                def wrapper(*args, **kwargs):
-                    """
-                    This is  a very ugly hack to get micropython working because
-                    `inspect.signature` doesn't exist. It may be actually better
-                    to not try any magic for now and raise the error.
-                    """
-                    try:
-                        return func(*args, **kwargs)
-
-                    except TypeError as e:
-                        if "takes" in str(e) and "positional arguments" in str(e):
-                            return func()
-                        raise
-
+                wrapper = func
         else:
```

Comment:
```python
elif is_awaitable(func):
  # ...
else:
  # ...
```

maybe?

Reply:
This conditional (and related `else`) is to check the number of args the receiving function expects. 

TL;DR do you have:

```
def handler(event):
    ...
```

or 

```
def handler():
    ...
```

Inside each branch is where the check for `is_awaitable` happens. Of course, it could have been done the other way around... but I didn't want to change the original code, just remove the "hack" for MicroPython now it's not needed.